sstables/storage.cc

```diff
@@ -54,6 +55,8 @@ class filesystem_storage final : public sstables::storage {
     future<> touch_temp_dir(const sstable& sst);
     future<> move(const sstable& sst, sstring new_dir, generation_type generation, delayed_commit_changes* delay) override;
     future<> rename_new_file(const sstable& sst, sstring from_name, sstring to_name) const;
+    future<> sync_cached_directory(std::string_view name) const;
+    friend future<> sync_cached_directory(const filesystem_storage* fsstor, std::string_view name);
```

Comment:
could define the friend function right here. more compact this way.

Reply:
Certainly thought of that; wasn't sure what the "size limit" was, for inlining simple function definitions like that. Will update.

Reply:
... actually, I'd rather not do that. If I inline the friend function's definition, then it becomes inaccessible (impossible to reference from outside the class definition) -- function overloading (i.e., different parameter list) is not enough to distinguish the friend function anymore! Consider the following small reproducer:

```
class foo {
private:
    void f1();
    friend void f1(foo *f) {
        f->f1();
    }
    void f2();
};

void foo::f1()
{
}

void foo::f2()
{
    f1(this);      // error: too many arguments to function call, expected 0, have 1
    ::f1(this);    // error: no member named 'f1' in the global namespace
    foo::f1(this); // error: too many arguments to function call, expected 0, have 1
}
```

Thus, I either have to define the friend function separately, or rename it. Which one do you prefer?

Reply:
https://stackoverflow.com/a/8284809

Reply:
Note that this works (per stackoverflow explanation above):

```
class foo;
void f1(foo *f);

class foo {
private:
    void f1();
    friend void f1(foo *f) {
        f->f1();
    }
    void f2();
};

void foo::f1()
{
}

void foo::f2()
{
    ::f1(this);
}
```

However, while it does work, it's worse than simply keeping the friend function's definition totally out of the class! So I guess I'll stick with the friend func as-is.